ebpf/xdp_filter.c

```diff
@@ -45,6 +45,10 @@
 #define USE_PERCPU_HASH    1
 #define GOT_TX_PEER	1
 
+/* set to non 0 to load balance in hardware mode on RSS_QUEUE_NUMBERS queues
+ * and unset BUILD_CPUMAP */
+#define RSS_QUEUE_NUMBERS   40
```

Comment:
would there be a way to make this not hardcoded?

Reply:
In theory yes, we can use a map that stores the number of threads and eBPF code could then get the number of threads from it. Problem is that this setup is specific to hardware bypass. So we could have something working by:
- making the build of this map conditional to harware offload
- get Suricata set the map if ever it is defined in the code

I'm wondering if I should introduce a xdp_hw_filter.c that would be usable without modification for hardware offload. I'm a bit reluctant on that as from eBPF point of view, the C code is just configuration. But it is clearly not the mentality and use case for something like Suricata.